service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionService.java

```diff
@@ -238,6 +238,33 @@ public void addEmptyDirectionsForAssigneeNotInMap(Map<DirectionAssignee, List<El
             .collect(toList());
     }
 
+    /**
+     * Sets a boolean to be yes/no to hide/reveal fields that should be hidden in tab view.
+     *
+     * @param directions any list of directions.
+     * @return a list of directions that have hidden tab values modified.
+     */
+    public List<Element<Direction>> changeSdoTabView(List<Element<Direction>> directions, String status) {
+        return directions.stream()
+            .map(element -> element(element.getId(), element.getValue().toBuilder()
+                .tabView(status)
+                .build()))
+            .collect(toList());
+    }
```

Comment:
this could be a method on the Direction java object, i.e a setter for this status... I am not sure I am a fan of tabView field at all though

Reply:
as discussed, we found a better solution 💪